chia/consensus/blockchain.py

```diff
@@ -352,12 +355,15 @@ async def _reconsider_peak(
         # This is not a heavier block than the heaviest we have seen, so we don't change the coin set
         return None, None, []
 
-    async def get_removals_and_additions(self, block: FullBlock) -> Tuple[List[bytes32], List[Coin]]:
+    async def get_removals_and_additions(
+        self, block: FullBlock, npc_result: Optional[NPCResult] = None
+    ) -> Tuple[List[bytes32], List[Coin]]:
         if block.is_transaction_block():
             if block.transactions_generator is not None:
-                block_generator: Optional[BlockGenerator] = await self.get_block_generator(block)
-                assert block_generator is not None
-                npc_result = get_name_puzzle_conditions(block_generator, self.constants.MAX_BLOCK_COST_CLVM, False)
+                if npc_result is None:
+                    block_generator: Optional[BlockGenerator] = await self.get_block_generator(block)
+                    assert block_generator is not None
```

Comment:
What type of log output will we get if this assert fails?

@mariano54 @arvidn @Yostra

Reply:
assert with throw an exception I believe